demos/object_detection_demo/python/README.md

```diff
@@ -276,6 +276,72 @@ To avoid disk space overrun in case of continuous input stream, like camera, you
 
 >**NOTE**: Windows\* systems may not have the Motion JPEG codec installed by default. If this is the case, you can download OpenCV FFMPEG back end using the PowerShell script provided with the OpenVINO &trade; install package and located at `<INSTALL_DIR>/opencv/ffmpeg-download.ps1`. The script should be run with administrative privileges if OpenVINO &trade; is installed in a system protected folder (this is a typical case). Alternatively, you can save results as images.
 
+## Running with OpenVINO Model Server
+
+You can also run this demo with model served in [OpenVINO Model Server](https://github.com/openvinotoolkit/model_server).
+
+Model Server is distributed as docker image, so you can pull it from Docker Hub with:
+
+```
+docker pull openvino/model_server
+```
+
+Use instruction from "Preparing to Run" section to download and convert the model and then place model files in the following directory structure:
+
+```
+<model_directory>
+└── <version_directory>
+    ├── model.bin
+    └── model.xml
+```
+
+For example, for face detection model, it could look like this:
+
+```
+face_detection
+└── 1
+    ├── face-detection-retail-0004.bin
+    └── face-detection-retail-0004.xml
+```
+
+Now you can run OpenVINO Model Server:
+```
+docker run -d -u $(id -u):$(id -g) -v <model_directory>:/model -p 9000:9000 openvino/model_server:latest \
+--model_path /model --model_name object_detection --port 9000
+```
+
+Above command will mount `<model_directory>` path on the host filesystem to the model server container, so the OVMS can load and serve the model. 
+
+**Note**: `<model_directory>` should indicate model catalog (**not** the one with specific version).
+
+Once the server has been launched it will provide inference service with the model called "object_detection" on port 9000. 
+
+To run the demo, you would have to provide `--adapter ovms` option and modify `-m` parameter to indicate model inference service instead of the model files. Model parameter for OVMS adapter follows this schema:
+
+```<service_address>/models/<model_name>[:<model_option>]```
+
+- `<service_address>` - OVMS gRPC service address in form `<address>:<port>`
+- `<model_name>` - name of the target model (the one specified by `model_name` parameter in the model server startup command)
+- `<model_version>` *(optional)* - version of the target model (default: latest)
+ 
+ Assuming that model server is running on the same machine as the demo, for above `docker run` command the correct value of `-m` parameter would be:
+
+`localhost:9000/models/object_detection`
+
+**Note**: While using `--adapter ovms`, inference options like: `-nireq`, `-nstreams` `-nthreads` as well as device specification with `-d` will be ignored.
+These values can be modified in OpenVINO Model Server configuration. See [model server configuration parameters](https://github.com/openvinotoolkit/model_server/blob/main/docs/docker_container.md#configuration-parameters) for more details.
```

Comment:
Are you going to add a note about auto shape for model server configuration? It would be nice to have an additional paragraph about how to allow model reshaping, since many demos require it

Reply:
Should we add this information to all demos or just the ones that require reshaping? I can see that for example object detection demo is not reshaping the model (at least in my tests). I'm wondering, if when demo is not reshaping the model, adding such paragraph would only make user confused.

On the other hand, we can add `--shape auto` option to all demos (even if they don't do reshape) and then keep paragraph about reshaping in all of them. What do you think?

Reply:
You are right, the paragraph about reshape in demos, which don't apply it, would only confuse the users. Let's provide this information only in demos that require it (exactly in bert_named_entity_recognition, object_detection, bert_question_answering, deblurring demos).
Actually, for object_detection we have `CTPN` modelAPI wrapper, which applies reshaping via `--input_size` key. So, it makes sense to add info about reshape here, although this wrapper supports only one model.
 


> On the other hand, we can add --shape auto option to all demos (even if they don't do reshape) and then keep paragraph about reshaping in all of them

Do you mean that demos might be launched with a command line argument `--shape`? I think it's not a good practice to provide an option, which doesn't change the behavior of the application for both adapters (`Openvino` and `Ovms`), if the demo is not going to apply reshaping.

Reply:
> Do you mean that demos might be launched with a command line argument --shape? I think it's not a good practice to provide an option, which doesn't change the behavior of the application for both adapters (Openvino and Ovms), if the demo is not going to apply reshaping.

I mean running model server with this option like:
`docker run -d -u $(id -u):$(id -g) -v <model_directory>:/model -p 9000:9000 openvino/model_server:latest \
--model_path /model --model_name object_detection --port 9000 --shape auto`

The demo command will not be affected. Just OVMS will allow the model to reshape "on-demand".

Reply:
Well, in this case we can add `--shape auto` to readme of all demos

Reply:
We shouldn't keep a description of how to run the server itself. It's up to OVMS docs. We don't want to fall into a risk of distributing outdated description. That naturally removes the question about mentioning `--shape` arg.

CC @vladimir-dudnik

Reply:
@Wovchena We cannot remove the whole description. The explanation how to apply inference parameters and reshape, when a user deals with OVMS adapter, is required, because it's related with demos.

My suggestion is to create a README for OVMS adapter, located in `model_api/adapters`. The documentation should contain the sections "how to install OVMS" and "Running demos with OVMS adapter". 
The second section might refer to external OVMS documentation, which explains how to pull the docker, create the IR files and put it to the specific folder, how to run the docker image. Finally, there will be paragraphs about infer options, model reshaping etc., and the schema for `--model` parameter.

I suppose the description, related to OMZ, is not going to become outdated so often, while OVMS running can be updated, but in this case we will not keep a description of it. So, the ModelAPI demos will refer to this README to describe how to work with OVMS adapter.